salt/haproxy/init.sls

```diff
@@ -0,0 +1,101 @@
+{% set grafana_server_node = salt['pnda.get_hosts_for_role']('grafana')[0] + '.' +pillar['consul']['node'] + '.' + pnda_domain %}
+{% set jupyter_server_node = salt['pnda.get_hosts_for_role']('jupyter')[0] + '.' +pillar['consul']['node'] + '.' + pnda_domain %}
+{% set pnda_home_dir = pillar['pnda']['homedir'] %}
+{% set haproxy_config_dir = '/etc/haproxy' %}
+{% set haproxy_lib_dir = '/var/lib/haproxy' %}
+{% set haproxy_stats_file = 'stats' %}
+{% set pnda_mirror = pillar['pnda_mirror']['base_url'] %}
+{% set haproxy_version = pillar['haproxy']['release_version'] %}
+{% set haproxy_package = 'haproxy-' + haproxy_version + '.tar.gz' %}
+{% set package_server = pillar['packages_server']['base_uri'] %}
+
+haproxy-group:
+  group.present:
+    - name: haproxy
+
+haproxy-user:
+  user.present:
+    - name: haproxy
+    - groups:
+      - haproxy
+    - createhome: False
+    - system: True
+    - require:
+      - group: haproxy-group
+
+# add a new user for HAProxy to be run under
+haproxy-create_user:
+  user.present:
+    - name: haproxy
+    - gid_from_name: True
+    - groups:
+      - haproxy
+
+# download haproxy binary from mirror and extract
+haproxy-install:
+  archive.extracted:
+    - name: {{ pnda_home_dir }}
+    - source: {{ package_server }}/{{ haproxy_package }}
+    - source_hash: {{ package_server }}/{{ haproxy_package }}.sha512.txt
+    - archive_format: tar
+    - if_missing: {{ pnda_home_dir }}/haproxy-{{ haproxy_version }}
+
+# create soft link to haproxy dir
+haproxy-create_soft_link:
+  cmd.run:
+    - name: 'ln -s {{ pnda_home_dir }}/haproxy-{{ haproxy_version}} {{ pnda_home_dir }}/haproxy'
+    - unless: ls {{ pnda_home_dir }}/haproxy
+
+# create soft link to haproxy binary
+haproxy-create_binary_soft_link:
+  cmd.run:
+    - name: 'ln -s {{ pnda_home_dir }}/haproxy/haproxy /usr/sbin/haproxy'
+
+# create haproxy configuration directory
+haproxy-create_conf_dir:
+  file.directory:
+    - name: '{{ haproxy_config_dir }}'
+
+# create haproxy library directory
+haproxy-create_lib_dir:
+  file.directory:
+    - name: '{{ haproxy_lib_dir }}'
+
+# add haproxy statistics file
+haproxy-create_empty_stats_file:
+  cmd.run:
+    - name: 'touch {{ haproxy_lib_dir }}/stats'
+
+# create configuration using template
+haproxy-create_config:
+  file.managed:
+    - source: salt://haproxy/templates/haproxy.cfg.tpl
+    - name: {{ haproxy_config_dir }}/haproxy.cfg
+    - template: jinja
+    - defaults:
+        grafana_server_node: {{ grafana_server_node }}
+        jupyter_server_node: {{ jupyter_server_node }}
+        pnda_domain: {{ pnda_domain }}
+
+# create softlink to haproxy manual
+haproxy-install_man:
+  cmd.run:
+    - name: 'ln -s {{ pnda_home_dir }}/haproxy/doc/haproxy.1 /usr/share/man/man1/haproxy.1'
+
+# set up haproxy systemd service
+haproxy-systemd:
+  file.managed:
+    - name: /etc/init.d/haproxy
+    - source: {{ pnda_home_dir }}/haproxy/examples/haproxy.init
```

Comment:
'examples' here suggests they're expecting us to use it as a basis for a file we place ourselves?